### PR TITLE
internal/envoy: Enable gzip compression for grpc-web content types.

### LINKED ISSUE #4345: Enable gzip compression for additional content type(s)
If I wanted to enable compression for another content type, how should I proceed?

_Originally posted by @bourquep in https://github.com/projectcontour/contour/issues/731#issuecomment-1039496574_

### PR DESCRIPTION
**Pull request by bourquep**
When Envoy serves as a `grpc-web` gateway, HTTP requests/responses use one of the [documented content-type](https://github.com/grpc/grpc-web#wire-format-mode), for which gzip compression is _not_ enabled by [default in Envoy](https://www.envoyproxy.io/docs/envoy/v1.17.4/api-v2/config/filter/http/compressor/v2/compressor.proto#envoy-api-msg-config-filter-http-compressor-v2-compressor).

This MR adds the `grpc-web` content types to Envoy's default ones, so they are gzip-compressed by default.

Fixes #4345

Signed-off-by: Pascal Bourque <pascal@studyo.co>

### CODE REVIEW

**Review by youngnick — APPROVED** [review]
LGTM, thanks for this change @bourquep!

**Comment by youngnick** [comment]
@projectcontour/maintainers, anyone else want to review this, before I hit merge?

**Comment by bourquep** [comment]
> Should we include more options for grpc-web content types e.g. plain application/grpc-web
> 
> see https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2

Good point. I had based my MR on [this document](https://github.com/grpc/grpc-web#wire-format-mode), from the readme of the grpc-web JS runtime. But indeed, all the formats defined in the grpc-web spec should probably be included.

**Comment by vahid-sohrabloo** [comment]
@youngnick When can you merge and release this PR?

**Review by sunjayBhatia — APPROVED** [review]
looks good, I might make a follow up PR to shorten the test change a tad, just so we can reduce some duplication

**Pull request merged**
